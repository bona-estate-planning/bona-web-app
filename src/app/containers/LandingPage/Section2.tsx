/**
 *
 * Section2
 *
 */
import { useMediaQuery, useTheme } from '@material-ui/core';
import Box from '@material-ui/core/Box';
import Container from '@material-ui/core/Container';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import React, { memo } from 'react';
import { ReactComponent as PlaceholderImage } from './placeholder.svg';

interface Props {}

const useStyles = makeStyles(theme => {
  return {
    section: (isMd: boolean) => ({
      backgroundColor: theme.palette.grey[50],
      padding: theme.spacing(isMd ? 10 : 3.5, 0),
      textAlign: 'center',
    }),
    headline: {
      fontWeight: theme.typography.fontWeightBold,
      fontSize: theme.spacing(3.5),
      marginBottom: theme.spacing(3),
    },
    subtitle: {
      fontSize: theme.spacing(2.25),
      marginBottom: theme.spacing(3),
    },
    placeholderImages: {
      margin: theme.spacing(0, -3.5),
    },
    placeholderImage: {
      display: 'inline-block',
      margin: theme.spacing(0, 3.5),
    },
  };
});

export const Section2 = memo((props: Props) => {
  const theme = useTheme();
  const isMd = useMediaQuery(theme.breakpoints.up('md'));
  const classes = useStyles(isMd);

  const placeholderImageCount = 4;
  const placeholderImages = [...Array(placeholderImageCount)].map((_, i) => (
    <PlaceholderImage
      key={`placeholder-image-${i}`}
      className={classes.placeholderImage}
    />
  ));

  return (
    <Box className={classes.section}>
      <Container>
        <Typography component="h3" className={classes.headline}>
          How it works
        </Typography>
        <Typography component="p" className={classes.subtitle}>
          Description of what the value prop is. The description should fit in
          here somewhere. Make it sound really nice.
        </Typography>
        <Box
          display={{ xs: 'none', md: 'block' }}
          className={classes.placeholderImages}
        >
          {placeholderImages}
        </Box>
      </Container>
    </Box>
  );
});
