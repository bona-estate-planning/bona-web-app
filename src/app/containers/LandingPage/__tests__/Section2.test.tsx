import React from 'react';
import { render } from '@testing-library/react';
import { Section2 } from '../Section2';

describe('<Section2  />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Section2 />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
