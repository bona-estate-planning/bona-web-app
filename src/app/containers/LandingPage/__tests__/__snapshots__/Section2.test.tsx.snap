// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section2  /> should match snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-7 makeStyles-section-1 makeStyles-section-6"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h3
      class="MuiTypography-root makeStyles-headline-2 MuiTypography-body1"
    >
      How it works
    </h3>
    <p
      class="MuiTypography-root makeStyles-subtitle-3 MuiTypography-body1"
    >
      Description of what the value prop is. The description should fit in here somewhere. Make it sound really nice.
    </p>
    <div
      class="MuiBox-root MuiBox-root-8 makeStyles-placeholderImages-4"
    >
      <svg
        class="makeStyles-placeholderImage-5"
      >
        placeholder.svg
      </svg>
      <svg
        class="makeStyles-placeholderImage-5"
      >
        placeholder.svg
      </svg>
      <svg
        class="makeStyles-placeholderImage-5"
      >
        placeholder.svg
      </svg>
      <svg
        class="makeStyles-placeholderImage-5"
      >
        placeholder.svg
      </svg>
    </div>
  </div>
</div>
`;
